src/external/ACSAdapter/AMSFileManager.ts

```diff
@@ -0,0 +1,120 @@
+import FramedClient from "@microsoft/omnichannel-amsclient/lib/FramedClient";
+
+type FileMetadata = Record<string, string>;
+
+enum FilePermission {
+    READ,
+    WRITE
+}
+
+interface IUploadedFile {
+   fileId: string;
+   metadata?: FileMetadata;
+}
+
+interface IAttachment {
+    name: string;
+    contentType: string;
+    contentUrl: string;
+    thumbnailUrl?: string;
+}
+
+interface IFileUploadRequest extends IAttachment {
+    permissions?: PermissionsOptions;
+}
+
+interface PermissionsOptions {
+    users: string[];
+    permission: FilePermission;
+}
+
+class AMSFileManager {
+    private amsClient: FramedClient;
+
+    public constructor(amsClient: FramedClient) {
+        this.amsClient = amsClient;
+    }
+
+    public async uploadFiles(files: IFileUploadRequest[]): Promise<IUploadedFile[]> {
+        return Promise.all(files.map(async (file: IFileUploadRequest) => this.uploadFileToAMS(file))) as Promise<IUploadedFile[]>;
+    }
+
+    public async downloadFiles(files: IUploadedFile[]): Promise<File[]> {
+        return Promise.all(files.map(async (file: IUploadedFile) => this.downloadFileFromAMS(file))) as Promise<File[]>;
+    }
+
+    public async updatePermissions(): Promise<void> {
+        return undefined;
+    }
+
+    public getFileIds(metadata?: Record<string, string>): string[] | undefined {
+        try {
+            return JSON.parse(metadata?.amsReferences as string) as string[];
+        } catch {
+            return undefined;
+        }
+    }
+
+    public createFileIdProperty(fileIds: string[]): Record<string, string> | undefined {
+        try {
+            return {
+                amsReferences: JSON.stringify(fileIds)
+            } as Record<string, string>;
+        } catch {
+            return undefined;
+        }
+    }
+
+    public getFileMetadata(metadata?: Record<string, string>): FileMetadata[] | undefined {
+        try {
+            return JSON.parse(metadata?.amsMetadata as string);
+        } catch {
+            return undefined;
+        }
+    }
+
+    public createFileMetadataProperty(metadata: FileMetadata[]): Record<string, string> | undefined {
+        try {
+            return {
+                amsMetadata: JSON.stringify(metadata)
+            };
+        } catch {
+            return undefined;

```

Comment:
do we want to log errors here? How will we determine failure in these methods?

Reply:
I added telemetry for this